play-framework/async-http/test/controllers/HomeControllerTest.java

```diff
@@ -0,0 +1,103 @@
+package controllers;
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.Application;
+import play.inject.guice.GuiceApplicationBuilder;
+import play.libs.concurrent.Futures;
+import play.libs.ws.WSClient;
+import play.libs.ws.ahc.AhcCurlRequestLogger;
+import play.mvc.Result;
+import play.mvc.Results;
+import play.test.WithServer;
+
+import java.time.Duration;
+import java.util.OptionalInt;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.IntStream;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static play.mvc.Http.Status.SERVICE_UNAVAILABLE;
+
+public class HomeControllerTest extends WithServer {
+    private final Logger log = LoggerFactory.getLogger(HomeControllerTest.class);
+
+    @Override protected Application provideApplication() {
+        return new GuiceApplicationBuilder().build();
+    }
+
+    @Test public void givenMultipleRequestsWhenResponseThenLog() throws InterruptedException {
+        AtomicInteger completedReqs = new AtomicInteger(0);
+        OptionalInt optHttpsPort = testServer.getRunningHttpsPort();
+        String url;
+        int port;
+        if (optHttpsPort.isPresent()) {
+            port = optHttpsPort.getAsInt();
+            url = "https://localhost:" + port;
+        } else {
+            port = testServer.getRunningHttpPort().getAsInt();
+            url = "http://localhost:" + port;
+        }
```

Comment:
Maybe the `url` and `port` can be set up in the test setup `@Before`, rather than at the start of each test?

Reply:
done